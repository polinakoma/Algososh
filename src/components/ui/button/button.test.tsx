import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "./button";

describe('Button', () => {
    it('is empty button', () => {
        const button = renderer
        .create(<Button />)
        .toJSON();
        expect(button).toMatchSnapshot();
    });
    it('is button with text', () => {
        const button = renderer
        .create(<Button text="Текст"/>)
        .toJSON();
        expect(button).toMatchSnapshot();
    });
    it('is disable button', () => {
        const button = renderer
        .create(<Button disabled={true}/>)
        .toJSON();
        expect(button).toMatchSnapshot();
    });
    it('is loading button', () => {
        const button = renderer
        .create(<Button isLoader={true}/>)
        .toJSON();
        expect(button).toMatchSnapshot();
    });
    it('has a callback upon click', () => {
        window.alert = jest.fn();
        render(<Button text="Text" onClick={() => alert('Callback')}/>)
        const button = screen.getByText("Text");
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith('Callback');
    }); 
});