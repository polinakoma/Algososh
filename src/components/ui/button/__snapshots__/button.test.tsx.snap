// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;

exports[`Button is disable button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button is empty button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button is loading button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
